core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -47,18 +66,23 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
     }
 }
 
+// is File a could type to use here?
+// No way to get script name if file is not present, should add another property (val fileName: String)
 interface ScriptContents {
     val file: File?
     val annotations: Iterable<Annotation>
```

Comment:
Users are likely to call `.isEmpty` on it, in which case it's still probably fully computed eagerly.

I'm fine with either `Collection` or `Sequence` or `Stream`. `Iterable` looks weird though